packages/compass-home/src/components/home.tsx

```diff
@@ -0,0 +1,209 @@
+/** @jsx jsx */
+import { css, jsx } from '@emotion/react';
+import React, { useEffect, useRef, useState } from 'react';
+import DataService from 'mongodb-data-service';
+import toNS from 'mongodb-ns';
+import AppRegistry from 'hadron-app-registry';
+
+import Workspace from './workspace';
+import Namespace from '../types/namespace';
+import InstanceLoadedStatus from '../constants/instance-loaded-status';
+import getRoleOrNull from '../modules/get-role-or-null';
+
+const homeViewStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'stretch',
+  height: '100vh',
+});
+
+const homePageStyles = css({
+  display: 'flex',
+  flexDirection: 'row',
+  alignItems: 'stretch',
+  flex: 1,
+  height: '100%',
+  zIndex: 0,
+});
+
+const defaultNS: Namespace = {
+  database: '',
+  collection: '',
+};
+
+function Home({
+  appRegistry,
+}: {
+  appRegistry: AppRegistry;
+}): React.ReactElement | null {
+  const connectRole = useRef(getRoleOrNull(appRegistry, 'Application.Connect'));
+
+  const [dataService, setDataService] = useState<DataService | null>(null);
+  const [isDataLake, setIsDataLake] = useState(false);
+  const [namespace, setNamespace] = useState<Namespace>(defaultNS);
+
+  const [errorLoadingInstanceMessage, setErrorMessage] = useState<
+    null | string
+  >(null);
+  const [instanceLoadingStatus, setInstanceLoadingStatus] = useState(
+    InstanceLoadedStatus.LOADING
+  );
+
+  useEffect(() => {
+    // Setup listeners.
+    function onInstanceRefreshed(instanceInformation: {
+      errorMessage?: string;
+      instance?: {
+        dataLake?: {
+          isDataLake?: boolean;
+        };
+      };
+    }) {
+      if (instanceInformation.errorMessage) {
+        setErrorMessage(instanceInformation.errorMessage);
+        setInstanceLoadingStatus(InstanceLoadedStatus.ERROR);
+
+        return;
+      }
+
+      setInstanceLoadingStatus(InstanceLoadedStatus.LOADED);
+
+      if (instanceInformation.instance?.dataLake?.isDataLake) {
+        setIsDataLake(true);
+      } else {
+        setIsDataLake(false);
+      }
+    }
+    appRegistry.on('instance-refreshed', onInstanceRefreshed);
+
+    function onDataServiceConnected(
+      err: Error | undefined | null,
+      ds: DataService
+    ) {
+      const StatusAction = appRegistry.getAction('Status.Actions');
```

Comment:
I think we can type all these getAction/getRole/getComponent interfaces better using function overloads with constant string argument to provide an exact return type (similar to event emitters typing in Node.js) for the registry items that are used in the plugins we are moving to typescript. It should limit the need for using as keyword and give us generally a better overview of the appRegistry interfaces (which should be helpful when we are cleaning it up eventually). You also already need to write a type for this as assertion, so moving it to appRegistry types should not be too much additional work.

I agree that rewriting the whole app registry and typing every single registry item doesn't make sense in this scope, but for the things that are used in this PR I feel that we should try to type them better.

Reply:
Real good suggestion, thanks - Added a context for appRegistry and added some more descriptive get role / get component types.
@addaleax has some work in progress on making the appRegistry in ts, so I think it was good we didn't go for it here!